### PR TITLE
support for session transaction and transaction snapshot.

### PR DESCRIPTION
**Pull request by poonai**
added support for snapshot and session-wide transactions as per 
https://www.postgresql.org/docs/9.3/sql-set-transaction.html

### CODE REVIEW

File: tests/sqlparser_common.rs [hunk 1]
```diff
@@ -3586,7 +3586,7 @@ fn parse_set_transaction() {
     // TRANSACTION, so no need to duplicate the tests here. We just do a quick
     // sanity check.
     match verified_stmt("SET TRANSACTION READ ONLY, READ WRITE, ISOLATION LEVEL SERIALIZABLE") {
-        Statement::SetTransaction { modes } => assert_eq!(
+        Statement::SetTransaction { modes, .. } => assert_eq!(
```
Comment (alamb) [inline tests/sqlparser_common.rs:3589]: I think it would be valuable to validate the contents of `session` and `mode` here

**Review by alamb — APPROVED** [review]
Thank you @poonai  -- look like a nice addition

**Comment by alamb** [comment]
@poonai  looks like this PR needs to have `rust fmt` run against it to get the CI green

**Comment by poonai** [comment]
@alamb done and thanks for the project. :)

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
> @alamb done and thanks for the project. :)


🙇  thank you -- I am but a caretaker of the giants that came before me (@andygrove and @nickolay, @Dandandan and @benesch largely)

**Comment by alamb** [comment]
I pushed https://github.com/sqlparser-rs/sqlparser-rs/pull/379/commits/ce23ba05b86442c1e5fb1a7576768e249bd84651 to get the CI to run cleanly

**Comment by alamb** [comment]
Thanks again @poonai

**Pull request merged**